azkaban-common/src/main/java/azkaban/executor/container/ContainerCleanupManager.java

```diff
@@ -170,21 +172,28 @@ private boolean shouldIgnore(final ExecutableFlow flow, final Status status) {
    *
    * @param flow
    */
-  private void cancelFlowQuietly(ExecutableFlow flow) {
+  private void cancelFlowQuietly(ExecutableFlow flow, Status originalStatus) {
     try {
-      Status originalStatus = flow.getStatus();
       logger.info(
           "Cleaning up stale flow " + flow.getExecutionId() + " in state " + originalStatus
               .name());
       this.containerizedDispatchManager.cancelFlow(flow, flow.getSubmitUser());
-      ExecutionControllerUtils.restartFlow(flow, originalStatus);
     } catch (ExecutorManagerException eme) {
       logger.error("ExecutorManagerException while cancelling flow.", eme);
     } catch (RuntimeException re) {
       logger.error("Unexpected RuntimeException while finalizing flow during clean up." + re);
     }
   }
 
+  private void retryFlowQuietly(ExecutableFlow flow, Status originalStatus) {
+    try {
+      logger.info("Restarting cleaned up flow " + flow.getExecutionId());
+      ExecutionControllerUtils.restartFlow(flow, originalStatus);
```

Comment:
Validation of status to whether it is restartable or not is taken care of inside restartFlow so no validation needed here right?

Reply:
Yes, that logic is already part of ExecutionControllerUtils.restartFlow